dds/DCPS/DCPS_Utils.cpp

```diff
@@ -464,24 +463,73 @@ bool repr_to_encoding_kind(DDS::DataRepresentationId_t repr, Encoding::Kind& kin
   case DDS::XCDR2_DATA_REPRESENTATION:
     kind = Encoding::KIND_XCDR2;
     break;
+  case OpenDDS::DCPS::UNALIGNED_CDR_DATA_REPRESENTATION:
+    kind = Encoding::KIND_UNALIGNED_CDR;
+    break;
   default:
     return false;
   }
   return true;
 }
 
-DDS::DataRepresentationIdSeq get_effective_data_rep_qos(const DDS::DataRepresentationIdSeq& qos, bool reader)
+DCPS::String repr_to_string(const DDS::DataRepresentationId_t& repr)
 {
-  if (qos.length() == 0) {
-    DDS::DataRepresentationIdSeq ids;
-    ids.length(reader ? 2 : 1);
-    ids[0] = DDS::XCDR2_DATA_REPRESENTATION;
-    if (reader) {
-      ids[1] = DDS::XCDR_DATA_REPRESENTATION;
+  DCPS::String repr_string;
+#ifndef OPENDDS_SAFETY_PROFILE
+  std::stringstream ss;
+#endif
+  switch(repr) {
+  case DDS::XCDR_DATA_REPRESENTATION:
+    repr_string = "XCDR_DATA_REPRESENTATION";
+    break;
+  case DDS::XML_DATA_REPRESENTATION:
+    repr_string = "XML_DATA_REPRESENTATION";
+    break;
+  case DDS::XCDR2_DATA_REPRESENTATION:
+    repr_string = "XCDR2_DATA_REPRESENTATION";
+    break;
+  case OpenDDS::DCPS::UNALIGNED_CDR_DATA_REPRESENTATION:
+    repr_string = "UNALIGNED_CDR_DATA_REPRESENTATION";
+    break;
+#ifndef OPENDDS_SAFETY_PROFILE
+  default:
+    ss << repr;
+    repr_string = ss.str();
+#endif
```

Comment:
This doesn't need to be conditional.  Use the helper functions in `SafetyProfileStreams.h` to convert an int to a string portably.